app/helpers/collection_time_helper.rb

```diff
@@ -1,16 +1,26 @@
 module CollectionTimeHelper
+  def collection_start_year(date)
+    window_end_date = Time.zone.local(date.year, 4, 1)
+    date < window_end_date ? date.year - 1 : date.year
+  end
+
   def current_collection_start_year
-    today = Time.zone.now
-    window_end_date = Time.zone.local(today.year, 4, 1)
-    today < window_end_date ? today.year - 1 : today.year
+    collection_start_year(Time.zone.now)
   end
 
   def collection_start_date(date)
-    window_end_date = Time.zone.local(date.year, 4, 1)
-    date < window_end_date ? Time.zone.local(date.year - 1, 4, 1) : Time.zone.local(date.year, 4, 1)
+    Time.zone.local(collection_start_year(date), 4, 1)
   end
```

Comment:
Would it make more sense to leave this as is and get the `collection_start_year call` method `collection_start_date(date).year` instead? Less hardcoded numbers